third_party/amd/lib/TritonAMDGPUTransforms/StreamPipeline.cpp

```diff
@@ -1128,7 +1128,7 @@ struct PipelinePass : public TritonAMDGPUStreamPipelineBase<PipelinePass> {
     // This removes additional barrier but pingpong will add the barrier again.
     // So we should just not do it to get a better vmcnt in front of each
     // AsyncCopy.
-    if (useAsyncCopy) {
+    if (useAsyncCopy && !useF16BlockPingpong) {
```

Comment:
Is this PR adding support for other precisions  to use numStages=3, or only 16bit is allowed to use num_stages=3?